extensions/sns/e2e/tests/sns.bash

```diff
@@ -17,65 +17,42 @@ teardown() {
 }
 
 # The location of the SNS init config.
-SNS_CONFIG_FILE_NAME="sns.yml"
-SNS_CONFIG_FILE_V2_NAME="sns_v2.yml"
+SNS_CONFIG_FILE_NAME="sns_init.yaml"
 
-@test "sns-cli binary exists and is executable" {
-    run "$(dfx cache show)"/extensions/sns/sns-cli --help
-    assert_output --partial "Initialize, deploy and interact with an SNS"
-}
-
-@test "sns config create and validate fail outside of a project" {
-    run dfx sns config create
-    assert_failure
-    assert_output --partial 'Error: Cannot find dfx configuration file in the current working directory. Did you forget to create one?'
-    run dfx sns config validate
-    assert_failure
-    assert_output --partial 'Error: Cannot find dfx configuration file in the current working directory. Did you forget to create one?'
-}
-
-@test "sns config create creates a default configuration" {
-    dfx_new
-    run dfx sns config create
-    assert_success
-    assert_output --regexp "Created SNS configuration at: .*/sns.yml"
-    : "Check that the file exists..."
-    test -e sns.yml
-}
-
-@test "sns config validate approves a valid configuration" {
+@test "sns init-config-file validate approves a valid configuration" {
     dfx_new
     install_asset sns/valid
-    run dfx sns config validate
+    run dfx sns init-config-file validate
     assert_success
-    assert_output --partial 'SNS config file is valid'
+    assert_output '' # no output if the file is valid
 }
 
-@test "sns config validate identifies a missing key" {
+@test "sns init-config-file validate identifies a missing key" {
     dfx_new
     install_asset sns/valid
-    grep -v token_name "${SNS_CONFIG_FILE_NAME}" | sponge "$SNS_CONFIG_FILE_NAME"
-    run dfx sns config validate
+    grep -v transaction_fee "${SNS_CONFIG_FILE_NAME}" | sponge "$SNS_CONFIG_FILE_NAME"
```

Comment:
Not sure, but IIUC, this line makes an invalid sns configuration file. A comment that says this would be good.

tangent: I guess sponge is used instead of `>` because `>` immediately clobbers the file before the command (in this case, grep) has a chance to see it.

How does the file get restored afterwards? Does @test restore files automatically?

Reply:
Done!

Reply:
Done!

Reply:
I'm not sure why sponge was chosen, but I think the file that gets (intentionally) messed up is a fresh temporary one created by the `install_asset sns/valid` command, not the one actually in the repo. So the file doesn't need to be restored, just deleted when the test is done. I could be wrong though, I don't really understand that well how this testing framework works

Reply:
Here's what's going on:
- `install_asset sns/valid` copies all the files in https://github.com/dfinity/dfx-extensions/tree/main/extensions/sns/e2e/assets/sns/valid into the current directory
- the `grep -v .. | sponge` command removes a required field from the otherwise valid config file
- the test makes sure that `sns config validate` notices the missing key

The test uses sponge order to read a file and write to the same file in the same command pipeline.
There's no need to "restore" the file because every test runs in a temporary directory, which is deleted after the test

Reply:
Thank you @ericswanson-dfinity !